### DISCUSSION TITLE
My Wifi Access Point not having gateIds

### DISCUSSION THREAD

**Discussion by Abh4git** [omnetpp/omnetpp#1098] [Q&A]
#944  am trying to add a concept of WirelessLinks extending the DataRateChannel and represent the otherwise dynamic wireless links in NED file. But when it comes to addPorts in my ScheduleConfigurator it seems to crash at findLinkOut(findInterface(node, networkInterface)); or linkOut = findLinkOut(node, networkInterface->getNodeOutputGateId());. At the end of the day I dont get a valid remoteNode in the wireless case. One of the reasons I see is gateId is always -1 in wlan interface cases. Just curios how this works. Is there some detailed descriptions which can help and is the gateId only for ethernet? Not currently clear for me. Some ppointers would be very helpful. (void GateScheduleConfiguratorBase::addPorts(Input& input) const method in GateScheduleConfiguratorBase.cc , link related loop. Any pointers would be of great help! 

for (auto networkNode : input.networkNodes) {

        auto node = check_and_cast<Node *>(topology->getNodeFor(networkNode->module));
        for (auto port : networkNode->ports) {
            auto networkInterface = check_and_cast<NetworkInterface *>(port->module); //It fails at remoteNode - Abhilash 20 Dec 2024
            EV_INFO << networkInterface->getDisplayName() << endl;
            auto link = findLinkOut(findInterface(node, networkInterface));
            EV_INFO << node->module->getDisplayName() << endl;
            auto linkOut = findLinkOut(node, networkInterface->getNodeOutputGateId());
            EV_INFO << node->module->getDisplayName() << endl;
            auto remoteNode = check_and_cast<Node *>(linkOut->getLinkOutRemoteNode());
            EV_INFO << remoteNode->module->getDisplayName() << endl;

**Comment by rhornig** [comment]
I don't get the goal here. You are trying to "extend" a wireless interface (which is a shared medium by definition) with something that extends DataRateChannel (which is inherently a point to point link)?

What do you want to achieve? Because the approach is wrong as far as I see.

**Comment by Abh4git** [comment]
Ok sorry, I will put it better. Thank you for looking into it. 1. Wireless like wifi is dynamic association happens by ssid or address association in config file2. I am trying to have a new link type called virtual link on top of DataRateChannel to represent the association in Ned file itself. 3. Having it in Ned file allows me to use it in scheduler is what I thought so that I can even see wireless links directly from topology4. Now I realise topology methods of findlinkOut, getRemoteNode might need some changes or I need to add new methods to help find these new type of links.I am trying to make some progress here. Only thing I still don't understand is getRemoteNode and it's real purpose for scheduling configuration is unclear to me. We need to get the source node and dest node. That I understand. Beyond that is unclear. 
I will continue looking on 4 and see if I can progress with the current details. 
ThanksRegardsAbhilash     On Sunday, January 21, 2024 at 11:16:50 PM GMT+5:30, Rudolf Hornig ***@***.***> wrote:  
 
 


I don't get the goal here. You are trying to "extend" a wireless interface (which is a shared medium by definition) with something that extends DataRateChannel (which is inherently a point to point link)?

What do you want to achieve? Because the approach is wrong as far as I see.

**Comment by Abh4git** [comment]
Thank you! I could find some solution. NetworkConfiguratorBase
NetworkConfiguratorBase::extractTopology seems to be the method which extracts topology from Ned file. Since my WirelessLink extending DataRateChannel is not really a link like the Wired Case, the source and destination interfaces we not getting updated. As of now I am going with a workaround that is in case of no interface found and its wireless then I get the first wireless interface (wlan0). This is for now working.
